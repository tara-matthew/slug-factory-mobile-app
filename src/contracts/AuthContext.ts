import {ILoginErrorResponse, ILoginSuccessResponse} from "./Login";

export interface IAuthContext {
    logout: () => void;
    onLogin: (username: string, password: string) => Promise<ILoginErrorResponse | ILoginSuccessResponse>;
    onRegister: (name: string, email: string, username: string, password: string, password_confirmation: string) => Promise<{ error: boolean}>;
}
