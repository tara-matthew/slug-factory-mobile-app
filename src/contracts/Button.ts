export interface IButtonProps {
    title: string;
    sendDataToParent: (data: never) => void;
}
