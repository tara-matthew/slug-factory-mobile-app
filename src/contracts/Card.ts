export interface ICardProps {
    item: {
        title: string;
    };
    imageURL: string;
    blurhash: string;
    sendDataToParent: (data: never) => void;
}
