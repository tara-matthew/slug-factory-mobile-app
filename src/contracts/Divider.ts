import {ReactNode} from "react";

export interface IDividerProps {
    width: DividerWidth;
}

export interface IContentWithDivider {
    top: boolean;
    bottom: boolean;
    outerClass?: string;
    innerClass?: string;
    children: ReactNode;
    dividerWidth: DividerWidth;
}

export enum DividerWidth {
    Half = 'half',
    Full = 'full',
}
