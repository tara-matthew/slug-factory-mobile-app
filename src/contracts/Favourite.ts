import { IPrint } from "./Print";

export interface IFavourite {
    id: string;
    type: string;
    favourited_at: string;
    resource: IPrint; // TODO union type with other favouritables
}
