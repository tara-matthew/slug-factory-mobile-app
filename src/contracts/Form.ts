import { IInputWithLabel } from "./InputWithLabel";

export interface IFormProps {
    inputs: IInputWithLabel[];
    buttonText: string;
    buttonTo: string;
    sendDataToParent: (data: { [value: string]: string }) => void;
}
