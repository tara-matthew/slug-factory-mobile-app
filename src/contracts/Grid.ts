import { IImage } from "./Image";

export interface IGridProps {
    items: IGridItem[];
}

export interface IGridItem {
    title: string;
    images: IImage[];
}
