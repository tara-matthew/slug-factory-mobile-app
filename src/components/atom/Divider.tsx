import React from "react";
import { StyleSheet, View } from "react-native";
import { DividerWidth, IDividerProps } from "../../contracts/Divider";

const Divider = (props: IDividerProps) => {
    const dividerClass = props.width === DividerWidth.Half ? "w-1/2" : "w-full";
    return (
        <View className={ dividerClass } style={ styles.divider } />
    );
};

const styles = StyleSheet.create({
    divider: {
        height: 1,
        backgroundColor: "gray",
    },
});

export default Divider;
