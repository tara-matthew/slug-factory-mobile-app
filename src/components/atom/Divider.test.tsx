import Divider from "./Divider";
import React from "react";
import { DividerWidth } from "../../contracts/Divider";
import { render } from "@testing-library/react-native";

it("renders correctly", () => {
    const widths = [DividerWidth.Half, DividerWidth.Full];

    widths.forEach((width) => {
        const { toJSON } = render(<Divider width={width} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
