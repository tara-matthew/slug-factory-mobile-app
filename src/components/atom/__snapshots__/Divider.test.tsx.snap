// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="w-1/2"
  style={
    {
      "backgroundColor": "gray",
      "height": 1,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<View
  className="w-full"
  style={
    {
      "backgroundColor": "gray",
      "height": 1,
    }
  }
/>
`;
